File: app/src/main/java/com/odin/movielist/Utils/MoviesRestAdapter.java
package com.odin.movielist.Utils;

import okhttp3.OkHttpClient;
import retrofit2.GsonConverterFactory;
import retrofit2.Retrofit;

/**
 * Created by mobil on 16.03.2017.
 */

public class MoviesRestAdapter {
    private static final String BASE_URL = "http://www.omdbapi.com/";
    private static Retrofit.Builder builder = new Retrofit.Builder().baseUrl(BASE_URL).addConverterFactory(GsonConverterFactory.create());
    private static Retrofit retrofit = builder.build();
    private static OkHttpClient.Builder httpClient =
            new OkHttpClient.Builder();
    public MoviesApi createService(){
        return retrofit.create(MoviesApi.class);
    }
}

File: app/src/main/java/com/odin/movielist/utils/MoviesRestAdapter.kt
package com.odin.movielist.utils

import okhttp3.OkHttpClient
import retrofit2.GsonConverterFactory
import retrofit2.Retrofit

/**
 * Created by Dino Omanovic on Mar 16, 2017
 */

class MoviesRestAdapter {
    fun createService(): MoviesApi {
        return retrofit.create(MoviesApi::class.java)
    }

    companion object {
        private val BASE_URL = "http://www.omdbapi.com/"
        private val builder = Retrofit.Builder().baseUrl(BASE_URL).addConverterFactory(GsonConverterFactory.create())
        private val retrofit = builder.build()
        private val httpClient = OkHttpClient.Builder()
    }
}

